core/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -336,7 +343,12 @@ private void balance() {
          * explain API only.
          */
         private MoveDecision decideRebalance(final ShardRouting shard) {
-            if (shard.started() == false) {
+            if (allocation.debugDecision()) {
+                if (shard.unassigned()) {
+                    // in debug mode, we can get a rebalance decision for all shards except unassigned ones
```

Comment:
`decideRebalance` is only called by `decideShardAllocation` which is only called by the explain API